Expecto.Differ/Library.fs

```diff
@@ -0,0 +1,45 @@
+module Expecto.Differ
+open System
+open DiffPlex.DiffBuilder
+open DiffPlex.DiffBuilder.Model
+
+let colorizedDiff (colorizeText: ConsoleColor -> string -> string) first second =
+  let first, second =
+    match box first, box second with
+    | (:? string as f), (:? string as s) ->
+      string f, string s
+    | f, s ->
+      sprintf "%A" f, sprintf "%A" s
+
+  let differ = SideBySideDiffBuilder(DiffPlex.Differ())
+  let diff = differ.BuildDiffModel(first, second)
+
+  let coloredText typ text =
+    match typ with
+    | ChangeType.Inserted -> colorizeText ConsoleColor.Green text
+    | ChangeType.Deleted -> colorizeText ConsoleColor.Red text
+    | ChangeType.Modified -> colorizeText ConsoleColor.Blue text
+    | ChangeType.Imaginary -> colorizeText ConsoleColor.Yellow text
+    | ChangeType.Unchanged | ChangeType.Imaginary | _ -> text
+
+  let colorizedDiff (lines: DiffPiece seq) =
+    lines
+    |> Seq.map (fun line ->
+      let styledLineNumber =
+        match line.Position |> Option.ofNullable with
+        | Some num ->
+          (string num).PadLeft(2)
+          |> coloredText line.Type
+        | None -> "~~~"
```

Comment:
Could you give your reasoning behind the `-` and `~` chars you use to deliniate parts of the diff?

Reply:
~~~ is now used for lines without a matching line on the other side. I guess a "gap" in the line numbering could work equally well. I wasn't really sure what it should look like, so this is just experimenting.

...I am leaning towards removing the whole line numbering/gutter rendering though. It's just too much imo – imo it's just as easy to read the diff without line numbers.